### PR TITLE
User controller

### PR DESCRIPTION
**Pull request by joredudiaz**
## Summary

- create user

## Trello Card

https://trello.com/c/L4tsAm8I

### CODE REVIEW

File: app/controllers/users.js [hunk 1]
```diff
@@ -0,0 +1,19 @@
+const { validationResult } = require('express-validator');
+const models = require('../../app/models');
+
+exports.create = (request, response) => {
+  const errors = validationResult(request);
```
Comment (molinajulian) [inline app/controllers/users.js]: This and the next line must be in a middleware, not in a controller.

File: app/controllers/users.js [hunk 2]
```diff
@@ -0,0 +1,19 @@
...
+  if (!errors.isEmpty()) return response.status(422).json({ errors: errors.array() });
+
+  return models.users
```
Comment (molinajulian) [inline app/controllers/users.js]: You must create a service which will use a model to create a user. You can't call models in controllers.

File: app/controllers/users.js [hunk 3]
```diff
@@ -0,0 +1,19 @@
...
+    .create(request.body)
+    .then(model => {
+      const modelJSON = JSON.parse(JSON.stringify(model));
```
Comment (molinajulian) [inline app/controllers/users.js]: It's not necessary. A model is an object.

File: app/controllers/users.js [hunk 4]
```diff
@@ -0,0 +1,19 @@
...
+      delete modelJSON.password;
+
+      response.status(201).json(modelJSON);
```
Comment (molinajulian) [inline app/controllers/users.js]: Use `.send` instead of `.json`

File: app/controllers/users.js [hunk 5]
```diff
@@ -0,0 +1,19 @@
...
+    })
+    .catch(error => {
```
Comment (molinajulian) [inline app/controllers/users.js]: In the controller, you must call to `next` in `.catch`

File: package.json [hunk 6]
```diff
@@ -67,16 +67,19 @@
     }
   },
   "dependencies": {
+    "bcrypt": "^4.0.1",
```
Comment (molinajulian) [inline package.json]: You have already installed `bcryptjs`. Why do you add `bcrypt` ?

File: package.json [hunk 7]
```diff
@@ -67,16 +67,19 @@
...
     "bcryptjs": "^2.4.3",
     "body-parser": "^1.18.2",
     "express": "^4.16.2",
-    "jwt-simple": "^0.5.1",
-    "sequelize": "^5.10.1",
+    "express-group-routes": "^1.1.0",
```
Comment (molinajulian) [inline package.json]: Delete this unnecessary package. And revert the changes in this file, because you don't need update packages or install another package.

File: app/models/user.js [hunk 8]
```diff
@@ -1,27 +1,44 @@
...
+        primaryKey: true
+      },
+      name: {
+        type: DataTypes.STRING,
+        allowNull: false,
+        validate: {
+          notNull: true
+        }
+      },
+      surname: {
```
[the remark this answers is not in the document]
Comment (geogy24) [reply app/models/user.js]: `surname`is one word not two

File: app/middlewares/users.js [hunk 9]
```diff
@@ -0,0 +1,32 @@
+const { checkSchema } = require('express-validator');
+const bcrypt = require('bcrypt');
+
+exports.create = checkSchema({
+  name: {
+    exists: true
+  },
+  surname: {
```
[the remark this answers is not in the document]
Comment (geogy24) [reply app/middlewares/users.js]: `surname` is one word not two

File: app/controllers/users.js (hunk 3 above)
Comment (geogy24) [reply app/controllers/users.js]: but in this model I can't remove the password to not show it in the response, so I must parse to JSON to remove it

Comment (arinaldi118) [reply app/controllers/users.js]: You are right. model is an instance of sequelize. You can make a copy of `dataValues` using `spread operator` and the you can delete the password of the copy.
